### ISSUE TITLE
Channel offset operator not removing strips

### ISSUE THREAD

**Issue by RichardAntalik** [GDQuest/blender-power-sequencer#452]
In issue #447 I have provided patch, but unfortunately it wasn't quite correct and I did not test it. Sorry for that :(.
But also never trust any code no matter how good it looks :)

`byp.context.sequences` is just generated collection of strips with no API functions. it happens to have `remove()` function, but that is collection internal function for managing members.

So this is mildly annoying, that to use API function, you have to find actual `sequences` collection that belongs to meta strip.
Here is patch for that, I have actually tested, but don't trust me anyway.

Also sorry for delay, I forgot to save todo list so I almost forgot about this issue. It's best to report even small bugs on tracker.

```
diff --git a/power_sequencer/operators/utils/functions.py b/power_sequencer/operators/utils/functions.py
index 021dfd54..ea43ff7f 100644
--- a/power_sequencer/operators/utils/functions.py
+++ b/power_sequencer/operators/utils/functions.py
@@ -252,6 +252,25 @@ def slice_selection(context, sequences, range_block=0):
     return broken_selection


+def find_sequences_by_strip(sequences, strip):
+    for strip_iter in sequences:
+        if strip_iter == strip:
+            return sequences
+
+        if strip_iter.type == 'META':
+            meta_sequences = find_sequences_by_strip(strip_iter.sequences, strip)
+            if meta_sequences is not None:
+                return meta_sequences
+
+    return None
+
+
+
+def remove_strip(context, strip):
+    sequences = find_sequences_by_strip(context.scene.sequence_editor.sequences, strip)
+    sequences.remove(strip)
+
+
 def trim_strips(context, frame_start, frame_end, to_trim, to_delete=[]):
     """
     Removes the footage and audio between frame_start and frame_end.
@@ -291,7 +310,7 @@ def trim_strips(context, frame_start, frame_end, to_trim, to_delete=[]):
             s.frame_final_end = trim_start

     for s in to_delete:
-        bpy.context.sequences.remove(s)
+        remove_strip(context, s)

     for s in initial_selection:
         s.select = True

```

**Comment by NathanLovato** [comment]
In which version of blender are you having this bug? And would you have a list of reproducible steps for me?

I need to be able to reproduce the bug to test it; so far everything's working as expected for me here.

**Comment by RichardAntalik** [comment]
It's issue you have reported to me in https://blender.chat/channel/vse?msg=BPrmy8H3mXoTe2QKk

**Comment by NathanLovato** [comment]
Ah sorry Richard, I didn't read your nickname, it's early morning here. So the bug @GTaulats is facing didn't involve any meta strips, but that's good to know.

I just did a minimal test and indeed it seems using `bpy.context.scene.sequence_editor.sequences` as opposed to `bpy.context.sequences` makes a difference.

@GTaulats would you try with just using `bpy.context.scene.sequence_editor.sequences` and see if it solves the issue you had?

**Comment by GTaulats** [comment]
Hi, in it's time I did a little research with the python console and it did work with `bpy.context.secene.sequence_editor.sequences`.

Then I also checked the state of the code and turns out the format of the strips change when using the loop. Actual example:
The list "to_delete" contains 1 strip: [bpy.data.scenes['Scene'].sequence_editor.sequences_all["Color.002"]]
Then when iterating in its for loop: <bpy_struct, ColorSequence("Color.002") at 0x0000014447296A38>.

I noticed that using bpy.context.sequences with the list's format "bpy.data..." works, but not with "<bpy_struct..."
So my conclusion is that the for loop "opens" the value of the "bpy.data..." when iterating. I tried to search for a solution but without much success.
